[pkg/otlp] Deprecate `metrics.send_monotonic_counter` in favor of `metrics.sums.cumulative_monotonic_mode`

mx-psi opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Draft PRs should be prefixed with `[WIP]` in their title.

-->
### What does this PR do?

<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succint form, consider
  opening multiple pull requests instead of a single one.
-->

Add a new configuration option, `metrics.sums.cumulative_monotonic_mode`, to replace `metrics.send_monotonic_counter`.

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

Counterpart to open-telemetry/opentelemetry-collector-contrib#8490

### Additional Notes

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

I tried to make the code look as similar as possible to the Collector code, but I did not link to the Collector issue, since I think it could be confusing.


### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

#### Test one

Run the OTLP pipeline with `metrics.send_monotonic_counter` set to true or false:
```yaml
otlp_config:
  receiver:
    protocols:
      grpc:
  metrics:
    send_monotonic_counter: true|false
```

Send OTLP cumulative monotonic Sum metrics (e.g. by using an OpenTelemetry Counter instrument), and check that:
- [ ] Delta is calculated or not for these sums depending on the `send_monotonic_counter`
- [x] A warning is logged telling users to switch to `metrics.sum.cumulative_monotonic_mode`

#### Test two

Run the OTLP pipeline with `metrics.sums.cumulative_monotonic_mode` set to `to_delta` or `raw_value`:

```yaml
otlp_config:
  receiver:
    protocols:
      grpc:
  metrics:
    sums:
      cumulative_monotonic_mode: to_delta|raw_value
```

Send OTLP cumulative monotonic Sum metrics (e.g. by using an OpenTelemetry Counter instrument), and check that:
- [ ] Delta is calculated or not for these sums depending on the mode, matching the `send_monotonic_counter` behavior (true -> to_delta, false -> raw_value)
- [x] No deprecation warning is logged

#### Test three

Run the OTLP pipeline with both `metrics.send_monotonic_counter` and `metrics.sums.cumulative_monotonic_mode`:

```yaml
otlp_config:
  receiver:
    protocols:
      grpc:
  metrics:
    send_monotonic_counter: true|false
    sums:
      cumulative_monotonic_mode: to_delta|raw_value
```

Check that:
- [x] The OTLP pipeline fails to start with an error indicating that both settings should not be used at the same time

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

mx-psi committed:
[pkg/otlp] Deprecate `metrics.send_monotonic_counter` in favor of `metrics.sums.cumulative_monotonic_mode`

mx-psi committed:
Update configuration template and add env variable

djmitche reviewed (approved):
pkg/config looks good

mx-psi committed:
Delete renames.dat

mx-psi committed:
Address review comments

This does the same changes as open-telemetry/opentelemetry-collector-contrib@027a481f84f9b8204993c2a280145c189b640936
